components/TextFields/src/ContainedInputView/private/MDCContainedInputViewLabelAnimation.m

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCContainedInputViewLabelAnimation.h"
+
+#import "MaterialAnimationTiming.h"
+
+static const CGFloat kMDCContainedInputViewLabelAnimatorDefaultAnimationDuration = (CGFloat)0.15;
+
+@implementation MDCContainedInputViewLabelAnimation
+
++ (void)layOutLabel:(nonnull UILabel *)label
+                 state:(MDCContainedInputViewLabelState)labelState
+      normalLabelFrame:(CGRect)normalLabelFrame
+    floatingLabelFrame:(CGRect)floatingLabelFrame
+            normalFont:(nonnull UIFont *)normalFont
+          floatingFont:(nonnull UIFont *)floatingFont {
+  UIFont *targetFont;
+  CGRect targetFrame;
+  MDCAnimationTimingFunction mdcTimingFunction;
+  if (labelState == MDCContainedInputViewLabelStateFloating) {
+    targetFont = floatingFont;
+    targetFrame = floatingLabelFrame;
+    mdcTimingFunction = MDCAnimationTimingFunctionAcceleration;
+  } else {
+    targetFont = normalFont;
+    targetFrame = normalLabelFrame;
+    mdcTimingFunction = MDCAnimationTimingFunctionDeceleration;
```

Comment:
`mdcTimingFunction` doesn't appear to be used anywhere in this method. Should it be wired up below?